src/chainstate/stacks/boot/pox.clar

```diff
@@ -153,8 +169,33 @@
             )
         ;; no state at all
         none
-    )
-)
+    ))
+
+(define-private (check-caller-allowed)
+    (or (is-eq tx-sender contract-caller)
+        (let ((caller-allowed 
+                 ;; if not in the caller map, return false
+                 (unwrap! (map-get? allowance-contract-callers
+                                    { sender: tx-sender, contract-caller: contract-caller })
+                          false)))
+          ;; is the caller allowance expired?
+          (if (< burn-block-height (unwrap! (get until-burn-ht caller-allowed) true))
+              (begin (map-delete allowance-contract-callers
```

Comment:
Why delete?

Reply:
Because the entries are expired, so the entries can be deleted. Ultimately, it wouldn't make much difference if we let expired entries remain here.

Reply:
Unless we need to check for the absence of entries in `allowance-contract-callers`, I think we should avoid deleting.  It would make the contract cheaper to execute without affecting correctness.

Reply:
Okay, removed in 49605a7a39b61dd6a70d439d62039cc902be4e84